SR2023/2023-03-27-virtual-competition-wrapup.md

```diff
@@ -0,0 +1,22 @@
+---
+to: Student Robotics 2023
+subject: SR2023 Virtual Competition wrap-up
+---
+
+Hello,
+
+That's a wrap on the virtual competition! Congratulations to all teams who entered, we saw some great performances and very interesting matches. If you were unable to catch the event live, you can find it on our [YouTube channel](https://www.youtube.com/watch?v=mTJGOhwmLPo).
+
+The current state of our league table can be found [on our website](https://studentrobotics.org/comp/league). This is up-to-date with all the points earned in the virtual competition, as well as those from the challenges.
+
+{% if attended %}
+
+The logs from your virtual matches have been posted in your teams' Discord channels. If you'd like them emailed instead, just let us know.
```

Comment:
Sending the logs like this is a huge addition to the amount of work. Simply asking for it on the other hand is very simple. Sure, scripting it would be nice, but as we don't have it already written, this is the best way.

Reply:
For clarity: I feel strongly that teams who are either inactive in discord or who have previously requested that comms go via email should not have to explicitly request that their logs be sent via email. I realise this is more work, however the alternative will give those teams (especially those who have already requested it) a very poor impression of us.

Reply:
> Sure, scripting it would be nice, but as we don't have it already written, this is the best way.

It really is very simple with Python, for example: https://stackoverflow.com/a/3363254/67873

We have also in the past (several times in fact) in SR implemented automated emailing of teams, precisely to avoid this sort of worry (though admittedly without attachments). It's only in recent years that we went back to doing it all manually (which continues to confuse me given the faff that doing it manually involves).

Reply:
Duly scripted: https://gist.github.com/PeterJCLaw/1d53eee245992e28509e81070f0ccaf9.
I've tested that by sending an email to myself with an attachment. You'll need to create an [app password](https://support.google.com/accounts/answer/185833?hl=en) to use it, but it's otherwise pretty straightforward.

Reply:
I got the script working after a few bugfixes. However it seems Google are rate-limiting me. I got BPV and CLY sent, but that's all. 

If you want to take a go, feel free, otherwise I'll send them out without logs first thing tomorrow.

Reply:
Can we prioritise the teams who are not in Discord and/or who have asked for comms via email?